File: src/main/java/client/effects/user/SquibEffect.java
package client.effects.user;

import client.effects.AbstractEffect;
import client.effects.EffectType;
import util.packet.PacketWriter;

public class SquibEffect extends AbstractEffect {

    private final String path;

    public SquibEffect(String path) {
        this.path = path;
    }

    @Override
    public EffectType getType() {
        return EffectType.SQUIB_EFFECT;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        pw.writeMapleString(path);
    }
}

File: src/main/java/client/effects/user/SquibEffect.kt
package client.effects.user

import client.effects.AbstractEffect
import client.effects.EffectType
import util.packet.PacketWriter

class SquibEffect(val path: String) : AbstractEffect() {

    override val type: EffectType get() = EffectType.SQUIB_EFFECT

    override fun encodeData(pw: PacketWriter) {
        pw.writeMapleString(path)
    }
}